test/widgets/cards_test.dart

```diff
@@ -0,0 +1,344 @@
+import 'package:motis_mitfahr_app/util/locale_manager.dart';
+import 'package:motis_mitfahr_app/util/profiles/profile_widget.dart';
+import 'package:motis_mitfahr_app/util/profiles/reviews/custom_rating_bar_indicator.dart';
+import 'package:motis_mitfahr_app/util/trip/drive_card.dart';
+import 'package:motis_mitfahr_app/util/trip/pending_ride_card.dart';
+import 'package:motis_mitfahr_app/util/trip/ride_card.dart';
+import 'package:motis_mitfahr_app/util/trip/seat_indicator.dart';
+
+import '../util/factories/drive_factory.dart';
+import '../util/factories/model_factory.dart';
+import '../util/factories/profile_factory.dart';
+import '../util/factories/profile_feature_factory.dart';
+import '../util/factories/ride_factory.dart';
+import '../util/mock_server.dart';
+import '../util/pump_material.dart';
+import '../util/request_processor.dart';
+import '../util/request_processor.mocks.dart';
+
+void main() {
+  final MockRequestProcessor processor = MockRequestProcessor();
+  late Ride ride;
+  late Drive drive;
+
+  setUpAll(() {
+    MockServer.setProcessor(processor);
+  });
+
+  setUp(() {
+    reset(processor);
+  });
+
+  //since the code is the same for RideCard DriveCard and PendingRideCard, we only test DriveCard
+  testWidgets('TripCard shows the correct information', (WidgetTester tester) async {
+    drive = DriveFactory().generateFake();
+    whenRequest(processor).thenReturnJson(drive.toJsonForApi());
+    await pumpMaterial(tester, DriveCard(drive, key: const Key('driveCard')));
+
+    //wait for card to load
+    await tester.pump();
+
+    //date is shown
+    expect(find.text(localeManager.formatDate(ride.startTime)), findsOneWidget);
+
+    //duration is shown
+    expect(find.byIcon(Icons.access_time_outlined), findsOneWidget);
+
+    //start location is shown
+    expect(find.byKey(const Key('start')), findsOneWidget);
+
+    //end location is shown
+    expect(find.byKey(const Key('end')), findsOneWidget);
+  });
+
+  group('PendingRideCard', () {
+    ride = RideFactory().generateFake(
+      status: RideStatus.pending,
+      rider: NullableParameter(ProfileFactory().generateFake(id: 1)),
+      seats: 1,
+    );
+    drive = DriveFactory().generateFake(rides: [ride], seats: 2);
+
+    setUp(() {
+      whenRequest(processor).thenReturnJson([]);
+    });
+
+    testWidgets('shows the correct information', (WidgetTester tester) async {
+      await pumpMaterial(tester, PendingRideCard(ride, reloadPage: () {}, drive: drive));
+
+      //Profile Widget from rider is shown
+      final Finder profileWidgetFinder = find.byType(ProfileWidget);
+      expect(profileWidgetFinder, findsOneWidget);
+      final ProfileWidget profileWidget = tester.widget(profileWidgetFinder);
+      expect(profileWidget.profile, ride.rider);
+
+      //Extra time is shown
+      expect(find.byKey(const Key('extraTime')), findsOneWidget);
+
+      //Seats indicator is shown
+      final Finder seatsIndicator = find.byType(IconWidget);
```

Comment:
Maybe better to check for an actual `SeatsIndicator` object.

Reply:
yes good idea thansk

Reply:
actually i just looked into the code, there is nor `SeatIndicator` object there, but a `IconWidget` with icons.chair that's why i test it that way. I can however change it to a `SeatIndicator` not sure if that's the best way. 
@Showcas can you explain why there is a `IconWidget` used here and no `SeatIndicator`?

Reply:
I think Simon and I forgot to change it there when we added `SeatIndicator`. Just change it as part of this PR, that should have been like that in the first place.

Reply:
yes but they do different things. `IconWidget` shows if the count is bigger than two only one icon and the count and `SeatIndicator` shows just the number of icons and i think the first one is better for a card

Reply:
Ah I see. In this case maybe we should still do this via the SeatIndicator with an optional boolean parameter (and that can just call IconWidget when it is true). But should be done in later refactorings of course.